velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp

```diff
@@ -0,0 +1,246 @@
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    BaseAggregate::template updateGroups<true, int64_t>(
+        groups,
+        rows,
+        args[0],
+        [](int64_t& result, int64_t value) {
+          if (result < value) {
+            result = value;
+          }
+        },
+        mayPushdown);
+  }
+
+  void addSingleGroupIntermediateResults(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    BaseAggregate::updateOneGroup(
+        group,
+        rows,
+        args[0],
+        [](int64_t& result, int64_t value) {
+          result = std::max(result, value);
+        },
+        [](int64_t& result, int64_t value, int /* unused */) {
+          result = value;
+        },
+        mayPushdown,
+        (int64_t)0);
+  }
+
+  void addSingleGroupRawInput(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdateSingleGroup(group, rows, args[0]);
+  }
+
+  void addRawInput(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdate(groups, rows, args[0]);
+  }
+
+ protected:
+  void updateOneAccumulator(
+      char* const group,
+      std::vector<vector_size_t>& rowSizes,
+      vector_size_t idx) {
+    if (decoded_.isNullAt(idx)) {
+      return;
+    }
+
+    // Clear null.
+    clearNull(group);
+    // Set max(current, this).
+    int64_t& current = *value<int64_t>(group);
+    current = std::max(current, (int64_t)rowSizes[idx]);
+  }
+
+  void
+  doUpdate(char** groups, const SelectivityVector& rows, const VectorPtr& arg) {
+    decoded_.decode(*arg, rows, true);
+
+    if (decoded_.isConstantMapping() && decoded_.isNullAt(0)) {
+      // There's nothing to do; all values are null.
+      return;
+    }
+
+    if (decoded_.isConstantMapping()) {
+      estimateSerializedSizes(arg, rows, 1);
+      rows.applyToSelected([&](vector_size_t i) {
+        updateOneAccumulator(groups[i], elementSizes_, 0);
+      });
+    } else {
+      estimateSerializedSizes(arg, rows, rows.countSelected());
+      rows.applyToSelected([&](vector_size_t i) {
+        updateOneAccumulator(
+            groups[i], elementSizes_, selectToElementSizesIndexMap_[i]);
+      });
+    }
+  }
+
+  // Estimate the sizes of first numToProcess selected elements in vector.
+  void estimateSerializedSizes(
+      VectorPtr vector,
+      const SelectivityVector& rows,
+      vector_size_t numToProcess) {
+    elementSizes_.resize(numToProcess);
+    std::fill(elementSizes_.begin(), elementSizes_.end(), 0);
+    selectToElementSizesIndexMap_.resize(rows.end() + 1);
+    elementIndices_.resize(numToProcess);
+    elementSizePtrs_.resize(numToProcess);
+    vector_size_t selectedIndex = 0;
+    auto selectivityIterator = SelectivityIterator(rows);
```

Comment:
use rows.applyToSelected

Reply:
applyToSelected might not be most apt for this because it applies to all selected indeces whereas we only want to apply to the first numToProcess selected indeces.